pallene/checker.lua

```diff
@@ -482,6 +482,11 @@ function Checker:check_stat(stat)
                 types.tostring(itertype), types.tostring(iteratorfn._type))
         end
 
+        if iteratorfn._tag == "ast.Exp.CallFunc" then
+            local iterfn_var = iteratorfn.exp.var._name
```

Comment:
Before we access the `.var` field we have to check that `iteratorfn.exp._tag` is "ast.Exp.ExpVar".